scio-parquet/src/main/scala/com/spotify/scio/parquet/avro/ParquetAvroIO.scala

```diff
@@ -87,23 +92,79 @@ final case class ParquetAvroIO[T: ClassTag: Coder](path: String) extends ScioIO[
     val resource =
       FileBasedSink.convertToFileResourceIfPossible(ScioUtil.pathWithShards(path))
     val prefix = StaticValueProvider.of(resource)
-    val usedFilenamePolicy =
-      DefaultFilenamePolicy.fromStandardParameters(prefix, null, params.suffix, false)
-    val destinations = DynamicFileDestinations.constant[T](usedFilenamePolicy)
+    val fileNamePolicy =
+      if (
+        params.windowedFilenameFunction != WriteParam.DefaultWindowedFilenameFunction ||
```

Comment:
I'd avoid comparing functions. FWIW they could behave the same and still fail equality. We should rename the 2 `Default*Function`s `Fallback*Function`s? Also see comments below, can we use a `Option[Either[WFn, Fn]]` here?

Also, if `windowedFilenameFunction` is required for windowed writes, we should check `data.internal.getWindowingStrategy != WindowingStrategy.globalDefault()` like you did on line 119 right?

So basically something like:

```
if (optFn.isDefined) {
  check collection is windowed
  windowedFilenameFn = optFn.get.left.orElse(fallbackWindowedFn)
  filenameFn = optFn.get.right.orElse(fallback...)
  createFilenamePolicy
} else {
  DefaultFilenamePolicy...
}
```

Reply:
I agree re: comparing functions; the Option[Either[]] approach makes things a lot easier. WRT the check that the user provides a windowed filename function when the collection is windowed, I'd prefer to defer to the exception thrown by the default function as that seems more concise?

Reply:
But that exception will only be throw after job submission and when runner starts writing? Checking early would prevent job from being submitted.

Reply:
That's a good point; I'll update the explicit check.